src/core/components/core-tooltip.tsx

```diff
@@ -44,18 +46,37 @@ export function ChartTooltip({
   getTooltipContent: getTooltipContentOverrides,
   api,
   i18nStrings,
+  renderDebounceDuration = 0,
 }: CoreChartProps.TooltipOptions & {
   i18nStrings?: BaseI18nStrings;
   getTooltipContent?: CoreChartProps.GetTooltipContent;
   api: ChartAPI;
 }) {
   const [expandedSeries, setExpandedSeries] = useState<ExpandedSeriesState>({});
   const tooltip = useSelector(api.tooltipStore, (s) => s);
-  if (!tooltip.visible || tooltip.group.length === 0) {
+  const [debouncedTooltip, setDebouncedTooltip] = useState<ReactiveTooltipState | null>(null);
```

Comment:
Can we e.g. move the code that subscribes to the tooltip data and does debouncing into a small utility hook above that component so to encapsulate the debouncing logic inside?